client/src/call_executor.rs

```diff
@@ -96,7 +96,10 @@ where
 			None,
 		)
 		.map(|(result, _, _)| result)?;
-		self.backend.destroy_state(state)?;
+		{
+			let _ = self.backend.get_import_lock().read();
```

Comment:
That is not correct. It drops immediately the lock!

It is also not required to create a sub context here, as the functions end after it anyway.

Reply:
> That is not correct. It drops immediately the lock!

How so? The `RwLockReadGuard` returned by `read` lives till the end of the block.

> It is also not required to create a sub context here, as the functions end after it anyway.

Here it actually makes sense because we don't want to hold the lock for the duration of `into_encoded`

Reply:
https://github.com/rust-lang/rust/issues/40096